### PR TITLE
Fix 4 configs involving ttrh builder

### PR DESCRIPTION
**Pull request by VinInn**
see also
https://its.cern.ch/jira/browse/CMSTRACK-118

it involves
SeedFromConsecutiveHitsCreator
Where I removed all existAs
SeedForPhotonConversion1Leg, MuonTrackLoader and GoodSeedProducer
where TTRHBuilder has been added to config

default config updated and tested

hlt to be done

note
GoodSeedProducer is the only producer
all others are just PSet
SeedFromConsecutiveHitsCreator used in SeedGeneratorFromRegionHitsEDProducer
SeedForPhotonConversion1Leg used in PhotonConversionTrajectorySeedProducerFromSingleLeg
MuonTrackLoader used in many MuonTrack producers

### CODE REVIEW

**Comment by VinInn** [comment]
@mtosi, @perrotta 
please have a look

**Comment by slava77** [comment]
Is this in a similar state as CCC #7673 , expected to break HLT steps?

**Comment by VinInn** [comment]
@slava77 : Yes, HLT will fail.

**Comment by perrotta** [comment]
@VinInn

As Martin already wrote in the thread of the similar #7673, for having HLT able to run with this PR we need to parse its configurations in ConfDB. Would those config changes be not backward compatible, as soon as we parse them in ConfDB we will not be able to use that ConfDB config for further HLT developments until this PR is allowed in a release/prerelease that we can use for HLT development. Of course, we don't want to stop developing HLT menus and configs until that release is out. Therefore, in that case we would need to know when this PR is expected to be allowed in a CMSSW release (supposedly 75X, isn't it?), and only at that time we will be able to parse the new parameters in the relevant config.

HOWEVER, in this particular case I verified that the explicit addition of the TTRHBuilder parameter in several configs, as done here, is not incompatible with the current CMSSW releases, which look for that parameter anyhow (but are somehow protected in case they don't find it). Therefore, I don't expect problems from including that parameter in ConfDB since now and continue running such modified HLT configurations also with the current development releases. Since that TTRHBuilder parameter addiion will be within a PSet it does not even need a re-parsing of ConfDB, as it can be simply added by hand to it.

What must be done on the HLT side is the following:
- Add (by hand) the correct TTRHBuider parameter in all TrackLoaderParameters PSet's nested in tracking modules
- Test that it complies with the current CMSSW and with CMSSW + this PR
- Dump the corresponding menu in CMSSW: it will then be compatible with both the current developing release and the next updates

Mia, can you take care of preparing the updated configuration in ConfDB? (We can agree the details offline)

A.

**Comment by mtosi** [comment]
sure
I'm currently out of the office,
so I can start working on this from Monday, I'm afraid...
hope this is fine
  mia

**Comment by perrotta** [comment]
Uhm... No, for HLT is not as simple as I thought at the beginning even for this PR.

The number of newly inserted parameters is larger than the simple TTRHBuilder, and I think that not all them belong to a PSet nested in the HLT config (although complicate multiple insertions in the Tracking  modules can easlily mislead when looking for which config to look at...). Given the complicateness of the Tracking configurations, it is not even straightforward to derive a specific customization to be used momentarily when dumping the HLT config from ConfDB in CMSSW (this is for @fwyzard that suggested something similar as a temporary solution...).

Moreover, there is also one parameter that got renamed ("SimpleMagneticFiled" vs "magneticField").

In conclusion, unless the HLT Tracking experts may invent some better solution, I don't think that we can do anything else that parsing in ConfDB the configuration of the modules modified in this PR, and carefully retuning for HLT their parameters afterwards.

**Comment by davidlange6** [comment]
closing in 74x - there is the same PR open in 75x (now the development release)

**Comment by Martin-Grunewald** [comment]
@VinInn 
and also re-open this one...

**Comment by boudoul** [comment]
@Martin-Grunewald Just for my education : why should this be reopened for 74X ? Thanks

**Comment by VinInn** [comment]
@davidlange6 , please reopen

**Comment by Martin-Grunewald** [comment]
@boudoul 
If we manage, we want this in 74X

**Comment by davidlange6** [comment]
reopened- but keep the ccc work separate from this work (where there is not yet agreement to take the change on the reco side)

**Comment by slava77** [comment]
-1

superseded by #8390
